src/core/bpf-devices.c

```diff
@@ -30,15 +30,6 @@ static int bpf_access_type(const char *acc) {
 }
 
 int cgroup_bpf_whitelist_device(BPFProgram *prog, int type, int major, int minor, const char *acc) {
-        struct bpf_insn insn[] = {
-                BPF_JMP_IMM(BPF_JNE, BPF_REG_2, type, 6), /* compare device type */
-                BPF_MOV32_REG(BPF_REG_1, BPF_REG_3), /* calculate access type */
-                BPF_ALU32_IMM(BPF_AND, BPF_REG_1, 0),
-                BPF_JMP_REG(BPF_JNE, BPF_REG_1, BPF_REG_3, 3), /* compare access type */
-                BPF_JMP_IMM(BPF_JNE, BPF_REG_4, major, 2), /* compare major */
-                BPF_JMP_IMM(BPF_JNE, BPF_REG_5, minor, 1), /* compare minor */
-                BPF_JMP_A(PASS_JUMP_OFF), /* jump to PASS */
-        };
```

Comment:
tried before, but some compiler/linker combos didn't like it since they include variables in the code

Reply:
But those macros are just structure definitions. We do plenty of array or structure initialization with non-const data. Is this somehow different?

Reply:
dunno, i just made the experience that some compiler refused it... but if it works now I am all for it.